### DISCUSSION TITLE
After a while, the message starts up again

### DISCUSSION THREAD

**Discussion by Wintreist** [ag2ai/faststream#1951] [Q&A] [answered]
Logs:
> 2024-12-01 13:13:23,879 DEBUG    -      | artifact_download | cab271efc8 - downloading_file: 54.58% download
> 2024-12-01 13:13:23,978 DEBUG    -      | artifact_download | cab271efc8 - downloading_file: 54.61% download
> 2024-12-01 13:13:24,125 DEBUG    -      | artifact_download | cab271efc8 - downloading_file: 54.64% download
> 2024-12-01 13:13:24,185 DEBUG    -      | artifact_download | cab271efc8 - downloading_file: 54.67% download
> 2024-12-01 13:13:24,283 DEBUG    -      | artifact_download | cab271efc8 - downloading_file: 54.71% download
> 2024-12-01 13:13:24,382 DEBUG    -      | artifact_download | cab271efc8 - downloading_file: 54.74% download
> 2024-12-01 13:13:24,399 INFO     -      | artifact_download | cab271efc8 - Received
> 2024-12-01 13:13:24,498 DEBUG    -      | artifact_download | cab271efc8 - downloading_file: 54.77% download
> 2024-12-01 13:13:24,588 DEBUG    -      | artifact_download | cab271efc8 - downloading_file: 54.80% download
> 2024-12-01 13:13:26,047 DEBUG    -      | artifact_download | cab271efc8 - downloading_file: 55.21% download
> 2024-12-01 13:13:26,085 DEBUG    -      | artifact_download | cab271efc8 - downloading_file: 55.24% download
> 2024-12-01 13:13:26,334 DEBUG    -      | artifact_download | cab271efc8 - downloading_file: 0.03% download
> 2024-12-01 13:13:26,442 DEBUG    -      | artifact_download | cab271efc8 - downloading_file: 55.27% download
> 2024-12-01 13:13:26,494 DEBUG    -      | artifact_download | cab271efc8 - downloading_file: 55.30% download
> 2024-12-01 13:13:26,620 DEBUG    -      | artifact_download | cab271efc8 - downloading_file: 0.06% download

I used a message broker to arrange file downloads. 
The problem is that the files are large and after a while, apparently thinking that something had happened, faststream gave the same message to another subscriber.
Is there anything we can do about it? Or maybe this is the case with my broker (RabbitMQ)?
upd1:
I measured the time for 1 more launch. The first time the message is received at 12:24:16, the next time at 12:27:12

**Comment by Lancetnik** [comment, marked as answer]
I think, RabbitMQ just thinks, that the consumer died and resend the message to another one.
Did you check Delivery Acknowledgement Timeout section - https://www.rabbitmq.com/docs/consumers#acknowledgement-timeout ?

**Reply by Wintreist** [reply]
Thank you. I ask you in Telegram chat and you help me.
You're right, consumer connection died. It's because I used _async def_ instead _def_ in sync code

**Reply by Lancetnik** [reply]
Sorry, just missed this discussion. It's hard to monitor all platforms
